src/Xhgui/Profiles.php

```diff
@@ -253,6 +253,32 @@ public function getAll($options = array())
     }
 
     /**
+     * Encodes a profile to avoid mongodb key errors.
+     * @param array $profile
+     *
+     * @return array
+     */
+    protected function encodeProfile($profile)
+    {
+        if (!is_array($profile)) {
+            return $profile;
+        }
+        $target = array(
+          '__encoded' => true,
+        );
+        foreach ($profile as $k => $v) {
+            if (is_array($v)) {
+                $v = $this->encodeProfile($v);
```

Comment:
why is this recursive call? do we have any other problematic keys than the content of `array_keys($profile['profile'])`?

Reply:
Are there any nested structures inside a profile that I have ignored?

Reply:
If this is the case that we do not need recursive checking, we should instead opt for an easier solution.

What do you think about something like 
```php
$replacedKeys = str_replace('.', '．', array_keys($a));
array_combine($replacedKeys, $a);
```
?

Here's a demonstration, if you're intersted

```php
<?php
$a = ['fx' => ['a' => 'xxx', 'b' => 2, 'c' => 3], 'quux' => ['f' => 7, 'cx' => 3]];
print_r(array_combine(str_replace('x', 'y', array_keys($a)), $a));
```

prints

```
Array
(
    [fy] => Array
        (
            [a] => xxx
            [b] => 2
            [c] => 3
        )

    [quuy] => Array
        (
            [f] => 7
            [cx] => 3
        )

)
```

Reply:
I'd also recommend defining the `.` and `．` as constants (UNICODE_PERIOD and UNICODE_FULLWIDTH_PERIOD) for example?